CodeStarterCamp_Week4/FitnessCenter.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  FitnessCenter.swift
+//  CodeStarterCamp_Week4
+//
+//  Created by 정선아 on 2022/05/10.
+//
+
+import Foundation
+
+enum InputError: Error {
+    case empty
+    case outOfValue
+    case invaildValue
+}
+
+class FitnessCenter {
+    let name: String
+    var targetBodyCondition: BodyCondition
+    var member: Person?
+    var listOfRoutine: [Routine]
+    
+    init(name: String, targetBodyCondition: BodyCondition, member: Person, listOfRoutine: [Routine]) {
+        self.name = name
+        self.targetBodyCondition = BodyCondition()
+        self.member = member
+        self.listOfRoutine = listOfRoutine
+        
+    }
+    
+    convenience init(name: String, listOfRoutine: [Routine]) {
+        self.init(name: name, targetBodyCondition: BodyCondition(), member: Person(name: "", bodyCondition: BodyCondition()), listOfRoutine: listOfRoutine)
+    }
+    
+    func exerciseRoutine() {
+            do {
+                self.member?.name = try writeName()
+                self.targetBodyCondition = try inputBodyCondition()
+            } catch InputError.empty {
+                print("입력되지 않았습니다. 다시 입력해주세요.")
+            } catch InputError.invaildValue {
+                print("없는 번호입니다. 다시 입력해주세요.")
+            } catch InputError.outOfValue {
+                print("값의 갯수가 적거나 큽니다.")
+            } catch {
+                print("알 수 없는 오류입니다. 다시 시도해주세요.")
+            }
+        tryRoutine()
+        }
+    
+    func writeName() throws -> String {
+        print("안녕하세요. \(name)피트니스 센터입니다. 회원님의 이름은 무엇인가요?")
+        guard let inputName = readLine() else { throw InputError.empty }
+        return inputName
+    }
```

Comment:
readLine() 메서드가 활성화 됐을 때, 아무 입력도 하지 않고 바로 `엔터` 누르면 어떻게 될까요?
그러면 `nil` 이 입력되는 게 아니라, `빈 문자열("")`이 들어가게 됩니다.

솜이 작성하신 코드는 nil 입력을 막을 수는 있어도, 빈 문자열("")을 걸러주진 못하고 있어요!
이걸 해결하려면, 간단하게 `guard문`을 하나 더 추가하면 됩니다. 😄 

```suggestion
    func writeName() throws -> String {
        print("안녕하세요. \(name)피트니스 센터입니다. 회원님의 이름은 무엇인가요?")
        
        // inputName 이 nil 인지 아닌지 먼저 검증
        guard let inputName = readLine() else {
            throw InputError.empty
        }
        
        // 여기까지 왔다는 건, nil 은 아니라는 소리인데, 그러면 빈 문자열("")인지 검증
        guard inputName.isEmpty == false else {
            throw InputError.empty
        }
        
        // 여기까지 왔다는 건, nil 도 아니고, 빈 문자열("")도 아니란 소리임
        return inputName
    }
```

Reply:
앗 이것도 문제였어요! 제가 까먹었나봐요... PR에 못 적었군요🥺